test/helpers/actionCreators.ts

```diff
@@ -1,71 +1,89 @@
+import { ThunkAction } from 'redux-thunk'
+
 import {
   ADD_TODO,
+  AddTodo,
   DISPATCH_IN_MIDDLE,
+  DispatchInMiddle,
   GET_STATE_IN_MIDDLE,
+  GetStateInMiddle,
   SUBSCRIBE_IN_MIDDLE,
+  SubscribeInMiddle,
   UNSUBSCRIBE_IN_MIDDLE,
+  UnsubscribeInMiddle,
   THROW_ERROR,
-  UNKNOWN_ACTION
+  ThrowError,
+  UNKNOWN_ACTION,
+  UnknownAction
 } from './actionTypes'
-import { Action, AnyAction, Dispatch } from '../..'
 
-export function addTodo(text: string): AnyAction {
+export function addTodo(text: string): AddTodo {
   return { type: ADD_TODO, text }
 }
 
-export function addTodoAsync(text: string) {
-  return (dispatch: Dispatch): Promise<void> =>
-    new Promise(resolve =>
+export function addTodoAsync(text: string): ThunkAction<any, any, {}, AddTodo> {
```

Comment:
Now that redux thunk is being imported, we can leverage the `ThunkAction` type to be much more explicit about what kind of action this is; thus, get properly-typed actions throughout this file.